src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -434,6 +434,10 @@ public int getQuorumSize(){
     //last proposed quorum verifier
     public QuorumVerifier lastSeenQuorumVerifier = null;
 
+    // Lock object that guard access to quorumVerifier and lastSeenQuorumVerifier.
+    private byte[] qvLock = new byte[0];
```

Comment:
It should be *final* if you want to use it as a lock object. Also, as constant it would be renamed as *QV_LOCK*.

nit: Also, we usually use a `Object`:
```
private final Object LOCK = new Object();
```
or a `ReentrantReadWriteLock` (more verbose tough, with try-finally, etc).